products/views.py

```diff
@@ -0,0 +1,44 @@
+import json
+
+from django.shortcuts       import render
+from django.http            import JsonResponse
+from django.views           import View
+from django.db.models       import Q
+
+from products.models        import Category, Product, ProductImage, SubCategory, ProductTag, Tag
+
+
+class ProductListView(View):
+    def get(self, request):
+        category_id       = request.GET.get('category_id')
+        sub_category_id   = request.GET.get('sub_category_id')
+
+        pagination   = int(request.GET.get('pagination', 0))
+        limit        = int(request.GET.get('limit', 4))
+        offset       = pagination * 4
+
+        if category_id or sub_category_id:
+            products = Product.objects.filter(
+                Q(category_id = category_id) | 
+                Q(sub_category_id = sub_category_id))[offset:offset+limit]              
+        else:
+            products = Product.objects.all()[offset:offset+limit]
+
+        product_list = [{
+            'id'               : product.id,
+            'name'             : product.name,
+            'hashtag'          : product.hashtag,
+            'option' : [{ 
+                'option_id'        : option.id,
+                'price'            : option.price,
+                'quantity'         : option.quantity,
+                'weight'           : option.weight,
+                } for option in product.productoption_set.all()],
```

Comment:
```suggestion
                } for option in product_options],
```

product.productoption_set.all() 을 위에 변수화 해서 사용하셔도 괜찮습니다!

변수화 해놓는다고 해서 실제로 저 쿼리셋을 호춣하지 않고, 실제로 동작할 때 호출하기 때문에 괜찮습니다.

Reply:
변수화하는 내용에 대해 알려주셔서 감사합니다!